GitHubCodespaces.ps1

```diff
@@ -0,0 +1,538 @@
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER CodespaceName
+        The name of the codespace.
+
+    .PARAMETER PassThru
+        Returns the updated GitHub Issue.  By default, this cmdlet does not generate any output.
+        You can use "Set-GitHubConfiguration -DefaultPassThru" to control the default behavior
+        of this switch.
+
+    .PARAMETER Wait
+        If present will wait for the codespace to stop.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .INPUTS
+        GitHub.Codespace
+
+    .OUTPUTS
+        GitHub.Codespace
+
+    .EXAMPLE
+        Stop-GitHubCodespace -Name vercellone-effective-goggles-qrv997q6j9929jx8
+
+    .LINK
+        https://docs.github.com/en/rest/codespaces/codespaces?apiVersion=2022-11-28#stop-a-codespace-for-the-authenticated-user
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        ConfirmImpact='Low')]
+    [OutputType({$script:GitHubCodespaceTypeName})]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSReviewUnusedParameter", "", Justification="PassThru is accessed indirectly via Resolve-ParameterWithDefaultConfigurationValue")]
+    param(
+        [Parameter(Mandatory,
+            ValueFromPipeline,
+            ValueFromPipelineByPropertyName)]
+        [string] $CodespaceName,
+
+        [switch] $PassThru,
+
+        [switch] $Wait,
+
+        [string] $AccessToken
+    )
+
+    process
+    {
+        Write-InvocationLog
+
+        $telemetryProperties = @{
+            'CodespaceName' = Get-PiiSafeString -PlainText $CodespaceName
+        }
+
+        $params = @{
+            'UriFragment' = "user/codespaces/$CodespaceName/stop"
+            'Method' = 'POST'
+            'Description' = "Stop Codespace $CodespaceName"
+            'AccessToken' = $AccessToken
+            'AcceptHeader' = $script:symmetraAcceptHeader
+            'TelemetryEventName' = $MyInvocation.MyCommand.Name
+            'TelemetryProperties' = $telemetryProperties
+        }
+
+        if (-not $PSCmdlet.ShouldProcess($CodespaceName, "Stop Codespace $CodespaceName"))
+        {
+            return
+        }
+
+        $result = (Invoke-GHRestMethod @params | Add-GitHubCodespaceAdditionalProperties)
+
+        if ($Wait.IsPresent)
+        {
+            $waitParams = @{
+                'CodespaceName' = $CodespaceName
+                'AccessToken' = $AccessToken
+            }
+
+            $result = Wait-GitHubCodespaceAction @waitParams
+        }
+
+        if (Resolve-ParameterWithDefaultConfigurationValue -Name PassThru -ConfigValueName DefaultPassThru)
+        {
+            return $result
+        }
+    }
+}
+
+function Wait-GitHubCodespaceAction
```

Comment:
I'm in agreement with you that this can remain an internal helper for the time being.